portfolio/src/main/webapp/script.js

```diff
@@ -17,18 +17,119 @@
 const authCheckURL = '/auth/check';
 const getCommentsURL = (limit) => `/comments?limit=${limit}`;
 const deleteAllCommentsURL = '/delete-comments';
+const postReactionsURL = '/posts/reactions';
+const userPostReactionURL = '/posts/user/react';
 
 const STATUS_OK = 200;
 
+const reacts = {
+  none: {
+    id: 'none',
+    solidIcon: 'far fa-thumbs-up',
+    outlineIcon: 'far fa-thumbs-up',
+    text: 'Like',
+    colour: '#9e9e9e'
+  },
+  like: {
+    id: 'like',
+    solidIcon: 'fas fa-thumbs-up',
+    outlineIcon: 'far fa-thumbs-up',
+    text: 'Like',
+    colour: '#0fa3b1'
+  },
+  love: {
+    id: 'love',
+    solidIcon: 'fas fa-heart',
+    outlineIcon: 'far fa-heart',
+    text: 'Love',
+    colour: '#ef5350'
+  },
+  wow: {
+    id: 'wow',
+    solidIcon: 'fas fa-surprise',
+    outlineIcon: 'far fa-surprise',
+    text: 'Wow',
+    colour: '#ffc107'
+  },
+  laugh: {
+    id: 'laugh',
+    solidIcon: 'fas fa-laugh-squint',
+    outlineIcon: 'far fa-laugh-squint',
+    text: 'Haha',
+    colour: '#ffc107'
+  }
+}
+
+const reactMap = new Map([
```

Comment:
What's the difference between reacts and reactMap? Why need both?

Reply:
Sorry, what functions? Both these maps contain the same values (shared between both because JS uses reference counting), and both these maps key by the name of the reaction. And they use different syntax of initializing the data / accessing the data.

Reply:
In JS `foo.bar` is the same as `foo['bar']`

Reply:
Oh okay I understand what you mean now, I wasn't aware that keys could be referenced from strings in that fashion for maps. I have changed this now.